nodepool/elements/virt-customize/zuul.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+set -x
+set -e
+
+# outside chroot
+if [ $# == 1 ]; then
+    dir="$1"
+    cp /var/lib/nodepool/.ssh/zuul_rsa.pub "$dir/tmp/authorized_keys"
+    cp $0 $dir/tmp/custom
+    exec chroot "$dir" /tmp/custom
+fi
+
+# inside chroot
+
+apt update -y
+
+apt install -y openssh-server sudo cloud-init cloud-initramfs-growroot cloud-utils
+
+apt install -y apt-transport-https ca-certificates gnupg2 curl locales
+
+curl -s https://download.docker.com/linux/debian/gpg > /tmp/docker.gpg
+apt-key add /tmp/docker.gpg
+
+if [[ $(cat /etc/debian_version) =~ ^9\. ]]; then
+    # debian 9
+    PKGS="rsync iproute git virtualenv python-pip python3-pip traceroute libpq-dev python2.7 libpython2.7-dev python-pkg-resources python3.5 libpython3.5-dev python3-pkg-resources sudo docker-ce docker-ce-cli containerd.io devscripts dirmngr build-essential"
+    debian_distro=stretch
+else
+    # debian 10
+    PKGS="rsync iproute2 git python-pip python3-pip traceroute libpq-dev python2.7 libpython2.7-dev python-pkg-resources python3.7 libpython3.7-dev python3-pkg-resources python-virtualenv python3-virtualenv sudo docker-ce docker-ce-cli containerd.io devscripts dirmngr build-essential"
+    debian_distro=buster
+fi
+
+cat > /etc/apt/sources.list.d/docker.list <<EOF
+deb [arch=amd64] https://download.docker.com/linux/debian $debian_distro stable
+EOF
+
```

Comment:
Maybe add `export DEBIAN_FRONTEND=noninteractive` before running apt to avoid any questions.

Reply:
good idea